packages/next/telemetry/events/version.ts

```diff
@@ -1,33 +1,85 @@
 const EVENT_VERSION = 'NEXT_CLI_SESSION_STARTED'
+const pageExtensions = ['tsx', 'ts', 'jsx', 'js']
 
-type EventCliSessionStarted = {
+type SessionStartedVersions = {
   nextVersion: string
   nodeVersion: string
+}
+
+type SessionStarted = {
   cliCommand: string
   isSrcDir: boolean | null
   hasNowJson: boolean
   isCustomServer: boolean | null
+  hasTypeScript: boolean
+}
+
+type SessionStartedNextConfig = {
```

Comment:
I'm not sure having these as separate types is valuable here.

Reply:
They are not really separated, the type `EventCliSessionStarted` still has all the fields, like before:

```ts
type EventCliSessionStarted = SessionStartedVersions &
  SessionStarted &
  SessionStartedNextConfig
```

The separation helped separating the types that I'm expecting to receive when the event is called, and the ones that the event is adding by itself, in summary it helps reducing the amount of types I have to write for the event here:

```ts
export function eventVersion(
  userConfig: { [key: string]: any } = {},
  event: SessionStarted &
    Pick<SessionStartedNextConfig, 'hasRewrites' | 'hasRedirects'>
)
```